### PR TITLE
[ADD] sale_coupon_order_line_link: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Technical module linking order lines with programs

cc @Tecnativa TT30850

### CODE REVIEW

File: sale_coupon_order_line_link/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Link to coupons to order lines",
```
Comment (pedrobaeza) [inline sale_coupon_order_line_link/__manifest__.py]:
```suggestion
    "name": "Link coupons to order lines",
```

File: sale_coupon_order_line_link/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "summary": "Adds a link between coupons and order lines to ease tracking",
```
Comment (pedrobaeza) [inline sale_coupon_order_line_link/__manifest__.py]:
```suggestion
    "summary": "Adds a link between coupons and their generated order lines for easing tracking",
```

File: sale_coupon_order_line_link/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        # https://git.io/J88As
+        vals = [r for r in res][0]
+        vals["sale_coupon_program_id"] = program.id
+        return {"x": vals}.values()
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    sale_coupon_program_id = fields.Many2one(
```
Comment (pedrobaeza) [inline sale_coupon_order_line_link/models/sale_order.py]: I think `sale` part is not needed (we are all in `sale` part):

```suggestion
    coupon_program_id = fields.Many2one(
```

File: sale_coupon_order_line_link/readme/DESCRIPTION.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+This module adds a link in the sale order line to the program used in a discount, so
+it can be easily tracked afterwards. Also eases the implementation of coupon modules
+that don't necessarly use the discount product as product for the discount line.
```
Comment (pedrobaeza) [inline sale_coupon_order_line_link/readme/DESCRIPTION.rst]:
```suggestion
that don't necessarily use the discount product as product for the discount line.
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
On the analysis, please add a filter called "Promotions" or similar for filtering those with the field set.

**Review by CarlosRoca13 — APPROVED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza All comments attended an added tests

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**